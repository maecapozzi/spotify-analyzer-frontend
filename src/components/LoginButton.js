import React from 'react'
import PropTypes from 'prop-types'
import injectSheet from 'react-jss'

const styles = {
  loginContainer: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'center',
    marginTop: '2em',
    textAlign: 'center'
  },
  loginButton: {
    backgroundColor: '#00C457',
    border: 'none',
    borderRadius: '30px',
    color: 'white',
    cursor: 'pointer',
    fontFamily: 'Inconsolata',
    fontSize: '1em',
    height: '4em',
    width: '20em',
    margin: [20, 20]
  },
  buttonLink: {
    fontFamily: 'Roboto',
    textDecoration: 'none'
  }
}

const LoginButton = ({ classes, url }) => {
  return (
    <div className={classes.loginContainer}>
      <a className={classes.buttonLink} href={`${url}/auth/spotify`}>
        <button className={classes.loginButton}>LOGIN WITH SPOTIFY</button>
      </a>
    </div>
  )
}

LoginButton.propTypes = {
  url: PropTypes.string
}

export default injectSheet(styles)(LoginButton)
