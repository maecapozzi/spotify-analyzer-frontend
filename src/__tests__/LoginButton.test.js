import React from 'react'
import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

import LoginButton from '../components/LoginButton'

configure({ adapter: new Adapter() })

test('it renders correctly', () => {
  const loginButtonProps = {
    url: 'http://localhost:3000/auth/spotify'
  }

  const component = shallow(<LoginButton {...loginButtonProps} />)
  expect(component).toMatchSnapshot()
})
