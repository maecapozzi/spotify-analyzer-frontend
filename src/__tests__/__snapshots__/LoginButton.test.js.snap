// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(LoginButton)
    url="http://localhost:3000/auth/spotify"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "buttonLink": "LoginButton-buttonLink-0-3",
        "loginButton": "LoginButton-loginButton-0-2",
        "loginContainer": "LoginButton-loginContainer-0-1",
      },
      "url": "http://localhost:3000/auth/spotify",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "buttonLink": "LoginButton-buttonLink-0-3",
          "loginButton": "LoginButton-loginButton-0-2",
          "loginContainer": "LoginButton-loginContainer-0-1",
        },
        "url": "http://localhost:3000/auth/spotify",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
